modules/data.land/R/soilgrids_soc_extraction.R

```diff
@@ -0,0 +1,209 @@
+  layer_thick <- c(0.05,0.10,0.15,0.30,0.40,1.00) # in unit m
+ 
+  # reproject locations to soilgrids projection
+  #Soilgrids data is using Homolosine projection https://www.isric.org/explore/soilgrids/faq-soilgrids 
+  p <- terra::vect(lonlat, crs = "+proj=longlat +datum=WGS84") # Users need to provide lon/lat
+  newcrs <- "+proj=igh +datum=WGS84 +no_defs +towgs84=0,0,0" 
+  p_reproj <- terra::project(p, newcrs) # Transform the point vector to data with Homolosine projection
+  
+  # setup progress bar
+  if (verbose) {
+    j <- 1
+    pb <- utils::txtProgressBar(min = 0, max = length(depths), char="*", width=70, style = 3)
+  }
+  
+  for (dep in seq_along(depths)) {
+    # setup virtual raster URLs for each layer
+    ocd_mean.url <- paste0(base_data_url,depths[dep],"_mean.vrt")
+    ocd_Q0.05.url <- paste0(base_data_url, depths[dep], "_Q0.05.vrt")
+    ocd_Q0.50.url <- paste0(base_data_url, depths[dep], "_Q0.5.vrt")
+    ocd_Q0.95.url <- paste0(base_data_url, depths[dep], "_Q0.95.vrt")
+    
+    # create virtual rasters && extract SOC values - the original unit is hg/m3
+    ocd_mean <- terra::extract(terra::rast(ocd_mean.url), p_reproj)
+    ocd_Q0.05_map <- terra::extract(terra::rast(ocd_Q0.05.url), p_reproj)
+    ocd_Q0.50_map <- terra::extract(terra::rast(ocd_Q0.50.url), p_reproj)
+    ocd_Q0.95_map <- terra::extract(terra::rast(ocd_Q0.95.url), p_reproj)
+    
+    #change the unit to more common kg/m3
+    ocd_mean_real <- ocd_mean[, -1] / 10
+    ocd_Q0.05_real <- ocd_Q0.05_map[, -1] / 10
+    ocd_Q0.50_real <- ocd_Q0.50_map[, -1] / 10
+    ocd_Q0.95_real <- ocd_Q0.95_map[, -1] / 10
+    
+    ocdquant[dep, ] <-c(ocd_mean_real,ocd_Q0.05_real,ocd_Q0.50_real,ocd_Q0.95_real)
+    ### Display progress to console
+    if (verbose) {
+      utils::setTxtProgressBar(pb, j)
+      j <- j+1
+      flush.console()}
+  
+  # cleanup interim results
+    rm(ocd_mean.url, ocd_Q0.05.url, ocd_Q0.50.url, ocd_Q0.95.url, 
+       ocd_mean, ocd_Q0.05_map, ocd_Q0.50_map, ocd_Q0.95_map,
+       ocd_mean_real, ocd_Q0.05_real, ocd_Q0.50_real, ocd_Q0.95_real)
+  }
+  
+  rownames(ocdquant) <- depths
+  colnames(ocdquant) <- c(rep("Mean", length(internal_site_info$site_info.lon)),
+                          rep("0.05", length(internal_site_info$site_info.lon)),
+                          rep("0.5", length(internal_site_info$site_info.lon)),
+                          rep("0.95", length(internal_site_info$site_info.lon)))
+  rownames(siteid) <- depths
+  colnames(siteid) <- c(rep("Mean", length(internal_site_info$site_info.lon)),
+                        rep("0.05", length(internal_site_info$site_info.lon)),
+                        rep("0.5", length(internal_site_info$site_info.lon)),
+                        rep("0.95", length(internal_site_info$site_info.lon)))
+  if (verbose) {
+    close(pb)
+  }
+  
+  # parse extracted data and prepare for output
+  ocd_fit <- reshape2::melt(ocdquant, id.vars = c("Mean"))
```

Comment:
Low priority: How difficult would it be to rewrite these manipulations to use something like `dplyr::pivot_longer` instead of adding reshape2 to the dependency list? data.land has a beastly number of dependencies already, so when there's a choice I recommend preferring solutions that use existing imports rather than add new ones.